docs/docs/migration-guide.mdx

```diff
@@ -24,6 +26,916 @@ If you still have training data in Markdown format then the recommended approach
+          domain: Domain,
+          precomputations: Optional[MessageContainerForCoreFeaturization] = None,
+          rule_only_data: Optional[Dict[Text, Any]] = None,
+          **kwargs: Any,
+      ) -> PolicyPrediction:
+          ...
+          tracker_state_features = self._featurize_tracker(
+              tracker, domain, precomputations, rule_only_data=rule_only_data
+          )
+          ...
+  ```
+
+  </TabItem>
+</Tabs>
+
+**Registering a Policy**
+
+Before you can use your custom policy you have to register your policy using the
+`DefaultV1Recipe.register` decorator. If your policy requires end-to-end features
+specify the graph component type `POLICY_WITH_END_TO_END_SUPPORT`. Otherwise, use
+`POLICY_WITHOUT_END_TO_END_SUPPORT`. Specify `is_trainable=True` if the `train`
+method of your policy should be called during the training. If your policy is only
+used during inference use `is_trainable=False`.
+
+
+<Tabs values={[{"label": "Rasa 2.0 (old)", "value": "old"}, {"label": "Rasa 3.0 (new)", "value": "new"}]}  groupId="3-0-policy-register" defaultValue="new">
+  <TabItem value="old">
+
+  ```python
+  class MyPolicy(Policy):
+      ...
+  ```
+
+  </TabItem>
+  <TabItem value="new">
+
+  ```python {3-5}
+  from rasa.engine.recipes.default_recipe import DefaultV1Recipe
+
+  @DefaultV1Recipe.register(
+      DefaultV1Recipe.ComponentType.POLICY_WITH_END_TO_END_SUPPORT,
+      is_trainable=True
+  )
+  class MyPolicy(Policy):
+      ...
+  ```
+
+  </TabItem>
+</Tabs>
+
+**Providing Rule-only Data to a Policy**
+
+Rasa Open Source allows excluding [forms](forms.mdx) or [slots](domain.mdx#slots) which
+are completely handled by
+[rules](rules.mdx) from becoming features in other policies.
+In Rasa Open Source 2 this information was passed onto the
+policies using the `set_shared_policy_states` method which set the policy attribute
+`_rule_only_data`. Rasa Open Source passes the names of rule-only slots and forms via the
+`predict_action_probabilities` method. The passed `rule_only_data` can be `None`
+in case the [`RulePolicy`](policies.mdx#rule-policy) is not part of your model
+configuration.
+
+<Tabs values={[{"label": "Rasa 2.0 (old)", "value": "old"}, {"label": "Rasa 3.0 (new)", "value": "new"}]}  groupId="3-0-policy-rule-only-data" defaultValue="new">
+  <TabItem value="old">
+
+  ```python
+  class MyPolicy(Policy):
+
+      def set_shared_policy_states(self, **kwargs: Any) -> None:
+          """Sets policy's shared states for correct featurization."""
+          self._rule_only_data = kwargs.get("rule_only_data", {})
+  ```
+
+  </TabItem>
+  <TabItem value="new">
+
+  ```python {6}
+  class MyPolicy(Policy):
+      def predict_action_probabilities(
+          self,
+          tracker: DialogueStateTracker,
+          domain: Domain,
+          rule_only_data: Optional[Dict[Text, Any]] = None,
+    ) -> PolicyPrediction:
+  ```
+
+  </TabItem>
+</Tabs>
+
+#### Changes to Custom NLU Components
```

Comment:
Is it more likely that folks are interested in porting an NLU component or a policy component? I don't have any data on this, but my gut thinks more folks want to move NLU components so maybe these should be mentioned first?